src/test/java/com/jcabi/email/EnvelopeTest.java

```diff
@@ -70,6 +74,42 @@ public void cachesPreviousCall() throws Exception {
     }
 
     /**
+     * Test for issue #8 (unicode/UTF-8 is broken).
+     * @throws Exception Thrown in case of problem of writing a message to
+     *  string.
+     */
+    @Test
+    public void handlesUnicodeCorrectly() throws Exception {
+        final Envelope env = new Envelope.MIME()
+            .with(new StSender("from <test-from@jcabi.com>"))
+            .with(new StRecipient("to", "test-to@jcabi.com"))
+            .with(new StSubject("test subject: test me"))
+            .with(new EnHTML("<html><body>привет</body></html>"));
+        final ByteArrayOutputStream stream =
+            new ByteArrayOutputStream();
+        env.unwrap().writeTo(stream);
+        final String msgtxt = stream.toString();
+        Assert.assertTrue(
```

Comment:
Use `MatcherAssert` and `Matchers` instead of `Assert` (see other tests),
same in lines below